lib/models/WebsiteConfiguration.ts

```diff
@@ -0,0 +1,215 @@
+export type Redirect = {
+    protocol?: string;
+    hostName?: string;
+    replaceKeyPrefixWith?: string;
+    replaceKeyWith?: string;
+    httpRedirectCode: string;
+};
+export type Condition = {
+    keyPrefixEquals?: string;
+    httpErrorCodeReturnedEquals?: string;
+};
+export type RoutingRuleParams = { redirect: Redirect; condition?: Condition };
+
+export class RoutingRule {
+    _redirect?: Redirect;
+    _condition?: Condition;
+
+    /**
+    * Represents a routing rule in a website configuration.
+    * @constructor
+    * @param params - object containing redirect and condition objects
+    * @param params.redirect - specifies how to redirect requests
+    * @param [params.redirect.protocol] - protocol to use for redirect
+    * @param [params.redirect.hostName] - hostname to use for redirect
+    * @param [params.redirect.replaceKeyPrefixWith] - string to replace
+    *   keyPrefixEquals specified in condition
+    * @param [params.redirect.replaceKeyWith] - string to replace key
+    * @param [params.redirect.httpRedirectCode] - http redirect code
+    * @param [params.condition] - specifies conditions for a redirect
+    * @param [params.condition.keyPrefixEquals] - key prefix that
+    *   triggers a redirect
+    * @param [params.condition.httpErrorCodeReturnedEquals] - http code
+    *   that triggers a redirect
+    */
+    constructor(params?: RoutingRuleParams) {
+        if (params) {
+            this._redirect = params.redirect;
+            this._condition = params.condition;
+        }
+    }
+
+    /**
+    * Return copy of rule as plain object
+    * @return rule;
+    */
+    getRuleObject() {
+        const rule = {
+            redirect: this._redirect,
+            condition: this._condition,
+        };
+        return rule;
+    }
+
+    /**
+    * Return the condition object
+    * @return condition;
```

Comment:
Is this condition a valid type?

Reply:
The type of `condition` is infered by TypeScript.